cluster-autoscaler/hack/update-deps.sh

```diff
@@ -48,22 +48,32 @@ MODS=($(
 popd
 rm -rf ${WORKDIR}
 
-for MOD in "${MODS[@]}"; do
-    V=$(
-        GOMOD="${MOD}@kubernetes-${VERSION}"
-        JSON=$(go mod download -json "${GOMOD}")
-        retval=$?
-        if [ $retval -ne 0 ]; then
-            echo "Error downloading module ${GOMOD}."
-            exit 1
-        fi
-        echo "${JSON}" | sed -n 's|.*"Version": "\(.*\)".*|\1|p'
-    )
-    go mod edit "-replace=${MOD}=${MOD}@${V}"
+PKGS=("." "./apis")
+for pkg in "${PKGS[@]}"; do
+  pushd "${pkg}"
+
+  for MOD in "${MODS[@]}"; do
```

Comment:
I think `MODS` depend on `pkg`, shouldn't `MOD` definition be moved inside the inner loop here? Also, perhaps move to a function to avoid too much nesting.

Reply:
> I think MODS depend on pkg

@x13n I think it's not correct since `MODS` is set based on the `staging` directory in the k/k repository.
So, we can not move the `MOD` definition inside the inner loop.

Reply:
`MODS` is a result of catting `go.mod`, which is different for each `PKG`. For instance, `./go.mod` uses `k8s.io/component-base`, but `./apis/go.mod` doesn't.

Reply:
Oh, it makes sense.
I'll try to fix it!

Reply:
@x13n Indeed, the `MODS` is the result of catting `go.mod`, but we cat the `go.mod` after we perform "pushed ${REPO}". It means that we cat the go.mod in k/k.

So, I think that the current script is correct. WDYT?

```
pushd ${REPO}
git fetch --depth 1 origin v${VERSION}
git checkout FETCH_HEAD
MODS=($(
    cat go.mod | sed -n 's|.*k8s.io/\(.*\) => ./staging/src/k8s.io/.*|k8s.io/\1|p'
))
```

Reply:
Anyway, I cleaned up this script: https://github.com/kubernetes/autoscaler/pull/6315#discussion_r1518672607